### PR TITLE
*: improve the format of the error log

### PR DESCRIPTION
**Pull request by nolouch**
Signed-off-by: nolouch <nolouch@gmail.com>

### What problem does this PR solve? 
Fix https://github.com/pingcap/tidb/issues/10586

- Only print stack information in `FATAL` level, the most of the time the stack information for error is useless.
- Use `ErrorVerbose` to control the stack information in error message  

### Check List 

Tests 

 - Unit test
before:
```
[2019/09/11 16:20:46.454 +08:00] [INFO] [store.go:68] ["new store"] [path=tikv:///tmp/tidb]
[2019/09/11 16:20:46.454 +08:00] [INFO] [client.go:148] ["[pd] create pd client with endpoints"] [pd-address="[]"]
[2019/09/11 16:20:46.454 +08:00] [ERROR] [client.go:204] ["[pd] failed to get cluster id"] [error="rpc error: code = Unavailable desc = all SubConns are in TransientFailure, latest connection error: connection error: desc = \"transport: Error while dialing dial tcp: missing address\""] [errorVerbose="rpc error: code = Unavailable desc = all SubConns are in TransientFailure, latest connection error: connection error: desc = \"transport: Error while dialing dial tcp: missing address\"\ngithub.com/pingcap/pd/client.(*client).getMembers\n\t/Users/nolouch/csn/program/go/pkg/mod/github.com/pingcap/pd@v0.0.0-20190712044914-75a1f9f3062b/client/client.go:239\ngithub.com/pingcap/pd/client.(*client).initClusterID\n\t/Users/nolouch/csn/program/go/pkg/mod/github.com/pingcap/pd@v0.0.0-20190712044914-75a1f9f3062b/client/client.go:201\ngithub.com/pingcap/pd/client.(*client).initRetry\n\t/Users/nolouch/csn/program/go/pkg/mod/github.com/pingcap/pd@v0.0.0-20190712044914-75a1f9f3062b/client/client.go:188\ngithub.com/pingcap/pd/client.NewClient\n\t/Users/nolouch/csn/program/go/pkg/mod/github.com/pingcap/pd@v0.0.0-20190712044914-75a1f9f3062b/client/client.go:161\ngithub.com/pingcap/tidb/store/tikv.Driver.Open\n\t/Users/nolouch/csn/program/go/src/github.com/pingcap/tidb/store/tikv/kv.go:80\ngithub.com/pingcap/tidb/store.newStoreWithRetry.func1\n\t/Users/nolouch/csn/program/go/src/github.com/pingcap/tidb/store/store.go:69\ngithub.com/pingcap/tidb/util.RunWithRetry\n\t/Users/nolouch/csn/program/go/src/github.com/pingcap/tidb/util/misc.go:46\ngithub.com/pingcap/tidb/store.newStoreWithRetry\n\t/Users/nolouch/csn/program/go/src/github.com/pingcap/tidb/store/store.go:67\ngithub.com/pingcap/tidb/store.New\n\t/Users/nolouch/csn/program/go/src/github.com/pingcap/tidb/store/store.go:51\nmain.createStoreAndDomain\n\t/Users/nolouch/csn/program/go/src/github.com/pingcap/tidb/tidb-server/main.go:235\nmain.main\n\t/Users/nolouch/csn/program/go/src/github.com/pingcap/tidb/tidb-server/main.go:176\nruntime.main\n\t/usr/local/Cellar/go/1.13/libexec/src/runtime/proc.go:203\nruntime.goexit\n\t/usr/local/Cellar/go/1.13/libexec/src/runtime/asm_amd64.s:1357"] [stack="github.com/pingcap/log.Error\n\t/Users/nolouch/csn/program/go/pkg/mod/github.com/pingcap/log@v0.0.0-20190307075452-bd41d9273596/global.go:42\ngithub.com/pingcap/pd/client.(*client).initClusterID\n\t/Users/nolouch/csn/program/go/pkg/mod/github.com/pingcap/pd@v0.0.0-20190712044914-75a1f9f3062b/client/client.go:204\ngithub.com/pingcap/pd/client.(*client).initRetry\n\t/Users/nolouch/csn/program/go/pkg/mod/github.com/pingcap/pd@v0.0.0-20190712044914-75a1f9f3062b/client/client.go:188\ngithub.com/pingcap/pd/client.NewClient\n\t/Users/nolouch/csn/program/go/pkg/mod/github.com/pingcap/pd@v0.0.0-20190712044914-75a1f9f3062b/client/client.go:161\ngithub.com/pingcap/tidb/store/tikv.Driver.Open\n\t/Users/nolouch/csn/program/go/src/github.com/pingcap/tidb/store/tikv/kv.go:80\ngithub.com/pingcap/tidb/store.newStoreWithRetry.func1\n\t/Users/nolouch/csn/program/go/src/github.com/pingcap/tidb/store/store.go:69\ngithub.com/pingcap/tidb/util.RunWithRetry\n\t/Users/nolouch/csn/program/go/src/github.com/pingcap/tidb/util/misc.go:46\ngithub.com/pingcap/tidb/store.newStoreWithRetry\n\t/Users/nolouch/csn/program/go/src/github.com/pingcap/tidb/store/store.go:67\ngithub.com/pingcap/tidb/store.New\n\t/Users/nolouch/csn/program/go/src/github.com/pingcap/tidb/store/store.go:51\nmain.createStoreAndDomain\n\t/Users/nolouch/csn/program/go/src/github.com/pingcap/tidb/tidb-server/main.go:235\nmain.main\n\t/Users/nolouch/csn/program/go/src/github.com/pingcap/tidb/tidb-server/main.go:176\nruntime.main\n\t/usr/local/Cellar/go/1.13/libexec/src/runtime/proc.go:203"]
[2019/09/11 16:20:47.460 +08:00] [ERROR] [client.go:204] ["[pd] failed to get cluster id"] [error="rpc error: code = Unavailable desc = all SubConns are in TransientFailure, latest connection error: connection error: desc = \"transport: Error while dialing dial tcp: missing address\""] [errorVerbose="rpc error: code = Unavailable desc = all SubConns are in TransientFailure, latest connection error: connection error: desc = \"transport: Error while dialing dial tcp: missing address\"\ngithub.com/pingcap/pd/client.(*client).getMembers\n\t/Users/nolouch/csn/program/go/pkg/mod/github.com/pingcap/pd@v0.0.0-20190712044914-75a1f9f3062b/client/client.go:239\ngithub.com/pingcap/pd/client.(*client).initClusterID\n\t/Users/nolouch/csn/program/go/pkg/mod/github.com/pingcap/pd@v0.0.0-20190712044914-75a1f9f3062b/client/client.go:201\ngithub.com/pingcap/pd/client.(*client).initRetry\n\t/Users/nolouch/csn/program/go/pkg/mod/github.com/pingcap/pd@v0.0.0-20190712044914-75a1f9f3062b/client/client.go:188\ngithub.com/pingcap/pd/client.NewClient\n\t/Users/nolouch/csn/program/go/pkg/mod/github.com/pingcap/pd@v0.0.0-20190712044914-75a1f9f3062b/client/client.go:161\ngithub.com/pingcap/tidb/store/tikv.Driver.Open\n\t/Users/nolouch/csn/program/go/src/github.com/pingcap/tidb/store/tikv/kv.go:80\ngithub.com/pingcap/tidb/store.newStoreWithRetry.func1\n\t/Users/nolouch/csn/program/go/src/github.com/pingcap/tidb/store/store.go:69\ngithub.com/pingcap/tidb/util.RunWithRetry\n\t/Users/nolouch/csn/program/go/src/github.com/pingcap/tidb/util/misc.go:46\ngithub.com/pingcap/tidb/store.newStoreWithRetry\n\t/Users/nolouch/csn/program/go/src/github.com/pingcap/tidb/store/store.go:67\ngithub.com/pingcap/tidb/store.New\n\t/Users/nolouch/csn/program/go/src/github.com/pingcap/tidb/store/store.go:51\nmain.createStoreAndDomain\n\t/Users/nolouch/csn/program/go/src/github.com/pingcap/tidb/tidb-server/main.go:235\nmain.main\n\t/Users/nolouch/csn/program/go/src/github.com/pingcap/tidb/tidb-server/main.go:176\nruntime.main\n\t/usr/local/Cellar/go/1.13/libexec/src/runtime/proc.go:203\nruntime.goexit\n\t/usr/local/Cellar/go/1.13/libexec/src/runtime/asm_amd64.s:1357"] [stack="github.com/pingcap/log.Error\n\t/Users/nolouch/csn/program/go/pkg/mod/github.com/pingcap/log@v0.0.0-20190307075452-bd41d9273596/global.go:42\ngithub.com/pingcap/pd/client.(*client).initClusterID\n\t/Users/nolouch/csn/program/go/pkg/mod/github.com/pingcap/pd@v0.0.0-20190712044914-75a1f9f3062b/client/client.go:204\ngithub.com/pingcap/pd/client.(*client).initRetry\n\t/Users/nolouch/csn/program/go/pkg/mod/github.com/pingcap/pd@v0.0.0-20190712044914-75a1f9f3062b/client/client.go:188\ngithub.com/pingcap/pd/client.NewClient\n\t/Users/nolouch/csn/program/go/pkg/mod/github.com/pingcap/pd@v0.0.0-20190712044914-75a1f9f3062b/client/client.go:161\ngithub.com/pingcap/tidb/store/tikv.Driver.Open\n\t/Users/nolouch/csn/program/go/src/github.com/pingcap/tidb/store/tikv/kv.go:80\ngithub.com/pingcap/tidb/store.newStoreWithRetry.func1\n\t/Users/nolouch/csn/program/go/src/github.com/pingcap/tidb/store/store.go:69\ngithub.com/pingcap/tidb/util.RunWithRetry\n\t/Users/nolouch/csn/program/go/src/github.com/pingcap/tidb/util/misc.go:46\ngithub.com/pingcap/tidb/store.newStoreWithRetry\n\t/Users/nolouch/csn/program/go/src/github.com/pingcap/tidb/store/store.go:67\ngithub.com/pingcap/tidb/store.New\n\t/Users/nolouch/csn/program/go/src/github.com/pingcap/tidb/store/store.go:51\nmain.createStoreAndDomain\n\t/Users/nolouch/csn/program/go/src/github.com/pingcap/tidb/tidb-server/main.go:235\nmain.main\n\t/Users/nolouch/csn/program/go/src/github.com/pingcap/tidb/tidb-server/main.go:176\nruntime.main\n\t/usr/local/Cellar/go/1.13/libexec/src/runtime/proc.go:203"]
```
 after:
```
[2019/09/11 16:19:38.544 +08:00] [INFO] [store.go:68] ["new store"] [path=tikv:///tmp/tidb]
[2019/09/11 16:19:38.545 +08:00] [INFO] [client.go:148] ["[pd] create pd client with endpoints"] [pd-address="[]"]
[2019/09/11 16:19:38.549 +08:00] [ERROR] [client.go:204] ["[pd] failed to get cluster id"] [error="rpc error: code = Unavailable desc = all SubConns are in TransientFailure, latest connection error: connection error: desc = \"transport: Error while dialing dial tcp: missing address\""]
[2019/09/11 16:19:39.549 +08:00] [ERROR] [client.go:204] ["[pd] failed to get cluster id"] [error="rpc error: code = Unavailable desc = all SubConns are in TransientFailure, latest connection error: connection error: desc = \"transport: Error while dialing dial tcp: missing address\""]
[2019/09/11 16:19:40.550 +08:00] [ERROR] [client.go:204] ["[pd] failed to get cluster id"] [error="rpc error: code = Unavailable desc = all SubConns are in TransientFailure, latest connection error: connection error: desc = \"transport: Error while dialing dial tcp: missing address\""]

```

### CODE REVIEW

File: config/config.go [hunk 1]
```diff
@@ -105,6 +106,9 @@ type Log struct {
 	Format string `toml:"format" json:"format"`
 	// Disable automatic timestamps in output.
 	DisableTimestamp bool `toml:"disable-timestamp" json:"disable-timestamp"`
+	// DisableErrorVerbose stops annotating logs with the full verbose error
+	// message.
+	DisableErrorVerbose bool `toml:"disable-error-verbose" json:"disable-error-verbose"`
```
Comment (zz-jason) [inline config/config.go]: how about `disable-error-stack`?

File: go.mod [hunk 2]
```diff
@@ -75,3 +75,5 @@ require (
 	sourcegraph.com/sourcegraph/appdash v0.0.0-20180531100431-4c381bd170b4
 	sourcegraph.com/sourcegraph/appdash-data v0.0.0-20151005221446-73f23eafcf67
 )
+
+go 1.13
```
Comment (zz-jason) [inline go.mod]: For now, the CI still use go 1.12.

File: config/config.go (hunk 1 above)
Comment (nolouch) [reply config/config.go]: Maybe better consistent with PD? Actually the field also called `errorVerbose` in zap, the main result of the full output with the stack due to the [`errors` package](https://godoc.org/github.com/pkg/errors#hdr-Formatted_printing_of_errors)

Comment (zz-jason) [reply config/config.go]: Yeah. But I think maybe `disable-error-stack` is easy to be understood by DBAs.

Comment (nolouch) [reply config/config.go]: ok

**Pull request merged**